File: src/main/scala/jkmau5/alternativeenergy/AltEngLog.java
package jkmau5.alternativeenergy;

import cpw.mods.fml.relauncher.FMLRelaunchLog;

import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * The main logger for AlternativeEnergy. Don't use System.out, because it's ugly, use this!
 *
 * @author jk-5
 */
public class AltEngLog {

    private static final Logger altEngLogger = Logger.getLogger("AlternativeEnergy");

    static {
        altEngLogger.setParent(FMLRelaunchLog.log.getLogger());
    }

    public static void log(Level level, String format, Object... data) {
        altEngLogger.log(level, String.format(format, data));
    }

    public static void log(Level level, Throwable ex, String format, Object... data) {
        altEngLogger.log(level, String.format(format, data), ex);
    }

    public static void severe(Throwable ex, String format, Object... data) {
        log(Level.SEVERE, ex, format, data);
    }

    public static void severe(String format, Object... data) {
        log(Level.SEVERE, format, data);
    }

    public static void warning(String format, Object... data) {
        log(Level.WARNING, format, data);
    }

    public static void warning(Throwable ex, String format, Object... data) {
        log(Level.WARNING, ex, format, data);
    }

    public static void info(String format, Object... data) {
        log(Level.INFO, format, data);
    }

    public static void fine(String format, Object... data) {
        log(Level.FINE, format, data);
    }

    public static void finer(String format, Object... data) {
        log(Level.FINER, format, data);
    }

    public static void finest(String format, Object... data) {
        log(Level.FINEST, format, data);
    }

    public static Logger getLogger() {
        return altEngLogger;
    }
}

File: src/main/java/jkmau5/alternativeenergy/AltEngLog.scala
package jkmau5.alternativeenergy

import java.util.logging.{Level, Logger}
import cpw.mods.fml.relauncher.FMLRelaunchLog

/**
 * No description given
 *
 * @author jk-5
 */
object AltEngLog {

  private final val altEngLogger = Logger.getLogger("AlternativeEnergy")
  this.altEngLogger.setParent(FMLRelaunchLog.log.getLogger)

  def log(level: Level, format: String, data: Any*) = this.altEngLogger.log(level, format.format(data: _*))
  def log(level: Level, ex: Throwable, format: String, data: AnyRef*) = this.altEngLogger.log(level, format.format(data: _*), ex)
  def severe(format: String, data: Any*) = this.log(Level.SEVERE, format, data)
  def severe(ex: Throwable, format: String, data: Any*) = this.log(Level.SEVERE, ex, format, data)
  def warning(format: String, data: Any*) = this.log(Level.WARNING, format, data)
  def warning(ex: Throwable, format: String, data: Any*) = this.log(Level.WARNING, ex, format, data)
  def info(format: String, data: Any*) = this.log(Level.INFO, format, data)
  def fine(format: String, data: Any*) = this.log(Level.FINE, format, data)
  def finer(format: String, data: Any*) = this.log(Level.FINER, format, data)
  def finest(format: String, data: Any*) = this.log(Level.FINEST, format, data)
  @inline def getLogger = this.altEngLogger
}
